curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/65ca2d0625aa3a3201067f70.md

```diff
@@ -0,0 +1,166 @@
+---
+id: 65ca2d0625aa3a3201067f70
+title: Step 30
+challengeType: 0
+dashedName: step-30
+---
+
+# --description--
+
+Inside the `array.forEach()` callback function, check if the current element is inside the `counts` object.  If the element can be found, update the value of `counts[el]` by `1`. Otherwise, assign the number `0` to `counts[el]`.
```

Comment:
```suggestion
Inside the `array.forEach()` callback function, check if the current element is inside the `counts` object.  If the element can be found, increment the value of `counts[el]` by `1`. Otherwise, assign the number `1` to `counts[el]`.
```

This doesn't work with `0` as the starting value when using the example code we gave in Step 26.

1. `0` is a falsy value.
2. If the number is used as a computed property name it means it was in the array at least once. So it might as well be set to `1`

---

This is the code example from the issue just for context to show the difference with code that would work with `0`

```
array.forEach((el) => {
  if (!counts[el]) {
    counts[el] = 0;
  }
  counts[el] = counts[el] + 1;
});
```